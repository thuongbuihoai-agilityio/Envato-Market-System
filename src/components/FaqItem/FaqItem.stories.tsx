import { StoryObj, Meta } from '@storybook/react';

// Components
import FaqItem from '@app/components/FaqItem';

const meta: Meta<typeof FaqItem> = {
  title: 'Custom Components/FaqItem',
  tags: ['autodocs'],
  component: FaqItem,
  argTypes: {
    question: {
      description: 'This is content of question',
    },

    answer: {
      description: 'This is content of answer',
    },
  },
};

type Story = StoryObj<typeof FaqItem>;

export const Default: Story = {
  args: {
    question: 'Question content',
    answer: 'Answer content',
  },
};

export default meta;
