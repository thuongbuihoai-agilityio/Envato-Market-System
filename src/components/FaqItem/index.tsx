import { AddIcon, MinusIcon } from '@chakra-ui/icons';
import { Box, Divider, Flex, Text } from '@chakra-ui/react';
import { useState } from 'react';

export type FaqItemProps = {
  question: string;
  answer: string;
};

const FaqItem = ({ question, answer }: FaqItemProps) => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleAnswer = () => {
    setIsOpen(!isOpen);
  };

  return (
    <Box>
      <Flex
        onClick={toggleAnswer}
        cursor="pointer"
        align="center"
        mb={5}
        pb={5}
        borderBottom="1px solid"
        borderColor="border.quinary"
        color="text.quinary"
      >
        {isOpen ? (
          <MinusIcon color="primary.500" />
        ) : (
          <AddIcon color="primary.500" />
        )}
        <Text ml={5} fontWeight="bold" color="text.quinary" fontSize="lg">
          {question}
        </Text>
      </Flex>
      {isOpen && (
        <Flex flex={1} pb={4}>
          <Divider
            orientation="vertical"
            height="auto"
            borderColor="primary.500"
            borderWidth="1px"
            borderRadius="lg"
            marginLeft={9}
          />
          <Flex>
            <Text color="secondary.1050" fontSize="sm" lineHeight="7" pl={4}>
              {answer}
            </Text>
          </Flex>
        </Flex>
      )}
    </Box>
  );
};

export default FaqItem;
