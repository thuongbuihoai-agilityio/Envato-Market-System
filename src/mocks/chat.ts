// Constants
import { AVATAR_POSITION, IMAGES } from '@app/constants';
import { MESSAGE_TIME } from '.';

// TODO: Update later
export const USER_CHATS = [
  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hi, I need more information',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 1000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 3000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 6000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Hi, i am here for help',
    avatarPosition: AVATAR_POSITION.AFTER,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 2000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 4000,
  },
  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 7000,
  },
];

export const MEMBER_CHATS = [
  { uid: 'admin', name: 'Admin' },
  { uid: 'user1', name: 'User 1' },
];
