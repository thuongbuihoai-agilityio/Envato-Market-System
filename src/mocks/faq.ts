export const FAQ_DATA = [
  {
    id: 1,
    question: 'What is the most important part of a dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 2,
    question: 'What are the three types of dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 3,
    question: 'What are examples of admin?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 4,
    question: 'What are 5 benefits of dashboards?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 5,
    question: 'What makes a good dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 6,
    question: 'What are the 3 basic skills of administrator?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
];
