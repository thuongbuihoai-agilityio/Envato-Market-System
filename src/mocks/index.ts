export * from './spending';
export * from './select';
export * from './table';
export * from './sidebar';
