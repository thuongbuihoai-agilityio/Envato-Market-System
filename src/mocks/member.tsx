import { Image } from '@chakra-ui/react';

// Constants
import { IMAGES } from '@app/constants';

// Utils
import { MESSAGE_TIME_FORMAT } from '.';

export const MEMBER_CHAT = [
  {
    id: 'user1',
    avatar: 'https://example.com/avatar1.jpg',
    name: 'Kien Nguyen',
    status: 'Online',
    localeTime: MESSAGE_TIME_FORMAT,
    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'online',
  },
  {
    id: 'user2',
    avatar: 'https://example.com/avatar2.jpg',
    name: 'Huy Pham',
    status: 'Away',
    localeTime: MESSAGE_TIME_FORMAT,

    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'offline',
  },
];
