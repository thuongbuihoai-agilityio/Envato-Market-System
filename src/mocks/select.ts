export const SOCIAL_PLATFORM_OPTIONS = [
  {
    value: 'github',
    label: 'github',
  },
  {
    value: 'facebook',
    label: 'facebook',
  },
  {
    value: 'instagram',
    label: 'instagram',
  },
];
