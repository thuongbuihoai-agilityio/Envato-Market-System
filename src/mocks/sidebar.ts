// constants
import {
  HELP_ITEM_LIST,
  MENU_ITEM_LIST,
  OTHER_ITEM_LIST,
} from '@app/constants/sidebar';

export const EXPAND_SIDEBAR_MENU_LIST = [
  {
    id: 1,
    title: 'Menu',
    listItem: MENU_ITEM_LIST,
  },
  {
    id: 2,
    title: 'Help',
    listItem: HELP_ITEM_LIST,
  },
  {
    id: 3,
    title: 'Others',
    listItem: OTHER_ITEM_LIST,
  },
];
