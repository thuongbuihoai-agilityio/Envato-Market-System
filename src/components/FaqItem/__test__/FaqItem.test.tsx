import { render, screen, fireEvent } from '@testing-library/react';

// Component
import FaqItem from '@app/components/FaqItem';

// Constants
import { FAQ_DATA } from '@app/mocks/faq';

test('renders question and answer correctly', () => {
  render(
    <FaqItem question={FAQ_DATA[0].question} answer={FAQ_DATA[0].answer} />,
  );

  // Assert that the question is rendered correctly
  const questionElement = screen.getByText(FAQ_DATA[0].question);
  expect(questionElement).toMatchSnapshot('question');

  // Assert that the answer is not initially visible
  const answerElement = screen.queryByText(FAQ_DATA[0].answer);
  expect(answerElement).toMatchSnapshot('answer');

  // Toggle the answer
  fireEvent.click(questionElement);

  // Assert that the answer is now visible
  const updatedAnswerElement = screen.getByText(FAQ_DATA[0].answer);
  expect(updatedAnswerElement).toMatchSnapshot('updated answer');
});
