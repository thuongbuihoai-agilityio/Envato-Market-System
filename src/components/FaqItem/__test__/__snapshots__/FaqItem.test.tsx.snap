// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders question and answer correctly: answer 1`] = `null`;

exports[`renders question and answer correctly: question 1`] = `
<p
  class="chakra-text css-xf6n75"
>
  What is the most important part of a dashboard?
</p>
`;

exports[`renders question and answer correctly: updated answer 1`] = `
<p
  class="chakra-text css-153g5n2"
>
  In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.
</p>
`;
